services/cloudEmail.js

```diff
@@ -0,0 +1,40 @@
+
+/*
+#     CloudBoost - Core Engine that powers Bakend as a Service
+#     (c) 2014 HackerBay, Inc. 
+#     CloudBoost may be freely distributed under the Apache 2 License
+*/
+
+
+var q = require('q');
+
+module.exports = function() {
+
+	return {
+		sendEmailsToAll: function(appId,emailBody,emailSubject){
+			var deferred = q.defer();
+			global.customService.find(appId,'User',null,{email:true},null,null,null,{},true).then(function(data){
+				if(data.length != 0){
+					var emailPromises = []
+					for(var k in data){
+						if(data[k].email){
+							emailPromises.push(global.mailService.emailCampaign(appId,data[k].email,emailBody,emailSubject))
+						}
+					}
+					q.all(emailPromises).then(function(data){
```

Comment:
Its a fire and forget function. dont wait for promises to complete and resolve it.